hermes/probe/create/create_file_test.go

```diff
@@ -0,0 +1,140 @@
+package create
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	metricpb "github.com/google/cloudprober/metrics/proto"
+	probepb "github.com/googleinterns/step224-2020/config/proto"
+	"github.com/googleinterns/step224-2020/hermes/probe"
+	"github.com/googleinterns/step224-2020/hermes/probe/metrics"
+	"github.com/googleinterns/step224-2020/hermes/probe/mock"
+	journalpb "github.com/googleinterns/step224-2020/hermes/proto"
+)
+
+func TestFileName(t *testing.T) {
+	tests := []struct {
+		file     RandomHermesFile
+		wantName string
+		wantErr  string
+	}{
+		{RandomHermesFile{51, 12}, "", "The file ID provided 51 wasn't in the required range [1,50]"},
+		{RandomHermesFile{0, 50}, "", "The file ID provided 0 wasn't in the required range [1,50]"},
+		{RandomHermesFile{3, 100}, "Hermes_03", ""},
+		{RandomHermesFile{12, 100}, "Hermes_12", ""},
+		{RandomHermesFile{3, 0}, "", "The file size provided 0 is not a positive number as required"},
+		{RandomHermesFile{3, 1001}, "", "The file size provided 1001 bytes exceeded the limit 1000 bytes"},
+	}
+
+	for _, test := range tests {
+		got, err := test.file.FileName()
+		if err == nil {
+			if test.wantErr != "" {
+				t.Errorf("{%v, %v}.FileName() gave wrong error expected %s, got nil", test.file.ID, test.file.Size, test.wantErr)
+			}
+			if test.wantName != got[0:9] {
+				t.Errorf("{%v, %v}.FileName() failed expected prefix %s, got %s", test.file.ID, test.file.Size, test.wantName, got[0:9])
+			}
+		} else {
+			if got != "" {
+				t.Errorf("{%v, %v}.FileName() failed expected empty string, got %s", test.file.ID, test.file.Size, got)
+			}
+			if err.Error() != test.wantErr {
```

Comment:
Requiring a concrete error message makes the tests brittle. These kinds of tests are called "change detection tests", which detect that the code was touched. It requires busy-work: any time you change the code, the tests have to be updated as well.
It's better to just check whether an error was returned. If the caller needs to distinguish between different error types, then also check that the correct error type is returned. Otherwise, just check whether `err` is nil/not-nil, as required by the test case. Just change the type of `wantErr` to `bool` to indicate whether an error is expected.

Reply:
Changed wantErr to bool